src/OpenLoco/CMakeLists.txt

```diff
@@ -0,0 +1,651 @@
+
+if (APPLE)
+    set(OLOCO_MM_SOURCES 
+        "${CMAKE_CURRENT_SOURCE_DIR}src/Platform/Platform.Macos.mm")
+    set_source_files_properties(${OLOCO_MM_SOURCES} PROPERTIES COMPILE_FLAGS "-x objective-c++")
+endif()
+
+# Disable optimizations for interop.cpp for all compilers, to allow optimized
+# builds without need for -fno-omit-frame-pointer
+if (MSVC)
+    set_source_files_properties("${CMAKE_CURRENT_SOURCE_DIR}/src/Interop/Interop.cpp" PROPERTIES COMPILE_FLAGS "/Oy- /Od")
+else ()
+    set_source_files_properties("${CMAKE_CURRENT_SOURCE_DIR}/src/Interop/Interop.cpp" PROPERTIES COMPILE_FLAGS "-fno-omit-frame-pointer -O0")
+endif ()
+
+if (MINGW)
+    loco_add_library(OpenLoco SHARED
+        PRIVATE_FILES
+            ${OLOCO_SOURCES})
+    target_link_libraries(OpenLoco winmm)
+else ()
+    if (MSVC)
+        # Headers added as well so that shown correctly in IDEs
+        loco_add_library(OpenLoco SHARED 
+            PRIVATE_FILES
+                ${OLOCO_SOURCES}
+                ${OLOCO_HEADERS})
+    else ()
+        loco_add_executable(OpenLoco 
+            PRIVATE_FILES
+                ${OLOCO_SOURCES}
+                ${OLOCO_MM_SOURCES})
+        add_dependencies(OpenLoco linkable_sections)
+        # set_source_files_properties has to be set in the CMakeLists that uses them
+        set_source_files_properties(
+            ${LOCO_SECTIONS}
+            PROPERTIES
+            EXTERNAL_OBJECT true
+            GENERATED true
+        )
+        target_sources(OpenLoco
+            PRIVATE
+                ${LOCO_SECTIONS})
+        set_target_properties(OpenLoco PROPERTIES LINK_FLAGS ${LOCO_SEGMENT_LINKER_FLAGS})
+
+        if (TARGET SDL2::SDL2main)
+            target_link_libraries(OpenLoco SDL2::SDL2main)
+        endif ()
+    endif ()
+endif ()
+
+target_link_libraries(OpenLoco
+    ${SDL2_LIB}
+    Threads::Threads
+    yaml-cpp::yaml-cpp
+    nonstd::span-lite
+    ghcFilesystem::ghc_filesystem
+    ${PNG_LIBRARY}
+    ${OPENAL_LIBRARIES})
+
+if (WIN32)
+    target_link_libraries(OpenLoco ${BREAKPAD_LIBRARIES})
+    target_link_libraries(OpenLoco winmm ws2_32 Resources)
+endif ()
+
+if (NOT WIN32)
+    add_dependencies(OpenLoco segfiles)
+endif()
+
+if (NOT APPLE AND NOT MSVC)
+    target_link_libraries(OpenLoco stdc++fs)
+endif()
+
+if (MINGW)
+    target_compile_definitions(OpenLoco PRIVATE DEBUG=${DEBUG_LEVEL} _WIN32_WINNT=0x0501 SDL_MAIN_HANDLED=1)
+    target_link_libraries(OpenLoco imm32 winmm version)
+    # newer GCC gets confused about our use of hardcoded address shenanigans
+elseif (MSVC)
+    target_compile_definitions(OpenLoco PRIVATE DEBUG=${DEBUG_LEVEL} _WINDLL _MBCS DEBUG_WIN32=1 _CRT_SECURE_NO_WARNINGS _CRT_NONSTDC_NO_DEPRECATE _USE_MATH_DEFINES SDL_MAIN_HANDLED _WINSOCK_DEPRECATED_NO_WARNINGS USE_BREAKPAD)
+else ()
+    target_compile_definitions(OpenLoco PRIVATE DEBUG=${DEBUG_LEVEL} _NO_LOCO_WIN32_=1)
+endif()
+
+if (APPLE AND FALSE) # TODO: This is broken after moving resources into src/Resources
```

Comment:
What's the plan for fixing that? Should it receive its own ticket?

Reply:
Its hard to fix atm whilst all the apple build is broken. When we can resurrect the apple build we can look into fixing this.